From 3854c061b9d4ff8cc613b777b100146b27b11706 Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Tue, 15 Oct 2019 18:02:52 +0800
Subject: [PATCH] net: rockchip_wlan: bcmdhd: fix gcc warning

drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c: In function 'sdioh_request_packet_chain':
drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c:1291:77: warning: '*((void *)&before+8)' may be used uninitialized in this function [-Wmaybe-uninitialized]
    write, ttl_len, now.tv_sec-before.tv_sec, now.tv_nsec/1000-before.tv_nsec/1000));
                                                                             ^
drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c:1107:23: note: 'before' was declared here
  struct timespec now, before;
                       ^~~~~~

Change-Id: I34e1b7132ad2771932e53475788dd4f3607f0233
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 .../rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c         | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c
index eafeb17f5c1a..cb54b832017e 100644
--- a/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c
+++ b/drivers/net/wireless/rockchip_wlan/rkwifi/bcmdhd/bcmsdh_sdmmc.c
@@ -835,7 +835,7 @@ sdioh_request_byte(sdioh_info_t *sd, uint rw, uint func, uint regaddr, uint8 *by
 #if defined(MMC_SDIO_ABORT)
 	int sdio_abort_retry = MMC_SDIO_ABORT_RETRY_LIMIT;
 #endif
-	struct timespec now, before;
+	struct timespec now, before = {};
 
 	if (sd_msglevel & SDH_COST_VAL)
 		getnstimeofday(&before);
@@ -1008,7 +1008,7 @@ sdioh_request_word(sdioh_info_t *sd, uint cmd_type, uint rw, uint func, uint add
 #if defined(MMC_SDIO_ABORT)
 	int sdio_abort_retry = MMC_SDIO_ABORT_RETRY_LIMIT;
 #endif
-	struct timespec now, before;
+	struct timespec now, before = {};
 
 	if (sd_msglevel & SDH_COST_VAL)
 		getnstimeofday(&before);
@@ -1104,7 +1104,7 @@ sdioh_request_packet_chain(sdioh_info_t *sd, uint fix_inc, uint write, uint func
 	uint8 *localbuf = NULL;
 	uint local_plen = 0;
 	uint pkt_len = 0;
-	struct timespec now, before;
+	struct timespec now, before = {};
 
 	sd_trace(("%s: Enter\n", __FUNCTION__));
 	ASSERT(pkt);
@@ -1302,7 +1302,7 @@ sdioh_buffer_tofrom_bus(sdioh_info_t *sd, uint fix_inc, uint write, uint func,
 {
 	bool fifo = (fix_inc == SDIOH_DATA_FIX);
 	int err_ret = 0;
-	struct timespec now, before;
+	struct timespec now, before = {};
 
 	sd_trace(("%s: Enter\n", __FUNCTION__));
 	ASSERT(buf);
@@ -1368,7 +1368,7 @@ sdioh_request_buffer(sdioh_info_t *sd, uint pio_dma, uint fix_inc, uint write, u
 {
 	SDIOH_API_RC status;
 	void *tmppkt;
-	struct timespec now, before;
+	struct timespec now, before = {};
 
 	sd_trace(("%s: Enter\n", __FUNCTION__));
 	DHD_PM_RESUME_WAIT(sdioh_request_buffer_wait);
-- 
2.35.3

